includes/class-wc-swedbank-plugin.php

```diff
@@ -0,0 +1,554 @@
+
+		// Add scripts and styles for admin
+		add_action( 'admin_enqueue_scripts', __CLASS__ . '::admin_enqueue_scripts' );
+
+		// Add Admin Backend Actions
+		add_action( 'wp_ajax_swedbank_pay_capture', array( $this, 'ajax_swedbank_pay_capture' ) );
+
+		add_action( 'wp_ajax_swedbank_pay_cancel', array( $this, 'ajax_swedbank_pay_cancel' ) );
+
+		// Filters
+		add_filter( 'swedbank_pay_generate_uuid', array( $this, 'generate_uuid' ), 10, 1 );
+		add_filter( 'swedbank_pay_payment_description', array( $this, 'payment_description' ), 10, 2 );
+
+		// Process swedbank queue
+		if ( ! is_multisite() ) {
+			add_action( 'customize_save_after', array( $this, 'maybe_process_queue' ) );
+			add_action( 'after_switch_theme', array( $this, 'maybe_process_queue' ) );
+		}
+
+		// Add admin menu
+		add_action( 'admin_menu', array( $this, 'admin_menu' ), 99 );
+
+		// Add Upgrade Notice
+		if ( version_compare( get_option( 'woocommerce_payex_psp_version', '1.2.0' ), '1.2.0', '<' ) ) {
+			add_action( 'admin_notices', __CLASS__ . '::upgrade_notice' );
+		}
+	}
+
+	public function includes() {
+		$vendors_dir = dirname( __FILE__ ) . '/../vendors';
+
+		//if ( ! class_exists( '\\SwedbankPay\\Api\\Client\\Client', false ) ) {
+		//require_once $vendors_dir.'/swedbank-pay-sdk-php/vendor/autoload.php';
+		//}
+
+		spl_autoload_register(
+			function ( $class_name ) {
+				$auto_load_path = dirname( __FILE__ ) . '/../vendors/swedbank-pay-sdk-php/src/';
+				$class_file     = $auto_load_path . str_replace( '\\', DIRECTORY_SEPARATOR, $class_name ) . '.php';
+				if ( file_exists( $class_file ) ) {
+					  require_once $class_file;
+
+					  return true;
+				}
+
+				return false;
+			}
+		);
+
+		spl_autoload_register(
+			function ( $class_name ) {
+				$auto_load_path = dirname( __FILE__ ) . '/../vendors/swedbank-pay-core-library/src/';
+				$class_file     = $auto_load_path . str_replace( '\\', DIRECTORY_SEPARATOR, $class_name ) . '.php';
+				if ( file_exists( $class_file ) ) {
+					  require_once $class_file;
+
+					  return true;
+				}
+
+				return false;
+			}
+		);
+
+		if ( ! class_exists( '\\Ramsey\\Uuid\\Uuid', false ) ) {
+			require_once $vendors_dir . '/ramsey-uuid/vendor/autoload.php';
+		}
+
+		require_once( dirname( __FILE__ ) . '/class-wc-swedbank-pay-transactions.php' );
+		require_once( dirname( __FILE__ ) . '/class-wc-swedbank-pay-icon.php' );
+	}
+
+	/**
+	 * Install
+	 */
+	public function install() {
+		// Install Schema
+		WC_Swedbank_Pay_Transactions::instance()->install_schema();
+
+		// Set Version
+		if ( ! get_option( 'woocommerce_payex_psp_version' ) ) {
+			add_option( 'woocommerce_payex_psp_version', '1.1.0' );
+		}
+	}
+
+	/**
+	 * Init localisations and files
+	 */
+	public function init() {
+		// Functions
+	}
+
+	/**
+	 * WooCommerce Init
+	 */
+	public function woocommerce_init() {
+		include_once( dirname( __FILE__ ) . '/class-wc-background-swedbank-pay-queue.php' );
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ["include_once" statement detected. File manipulations are discouraged. Statement is not a function, no parentheses are required. Concatenating is forbidden.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)